adapter/test/stdio/stdio_adapter_mode_test.cpp

```diff
@@ -0,0 +1,351 @@
+  if (set_path && strcmp(set_path, "1") == 0) {
+    auto paths = info.new_file + "," + info.existing_file;
+    setenv(kAdapterModeInfo, paths.c_str(), 1);
+  }
+  MPI_Init(argc, argv);
+  info.write_data = gen_random(args.request_size);
+  info.read_data = std::string(args.request_size, 'r');
+  return 0;
+}
+int finalize() {
+  MPI_Finalize();
+  return 0;
+}
+
+int pretest() {
+  if (fs::exists(info.new_file)) fs::remove(info.new_file);
+  if (fs::exists(info.new_file_cmp)) fs::remove(info.new_file_cmp);
+  if (fs::exists(info.existing_file)) fs::remove(info.existing_file);
+  if (fs::exists(info.existing_file_cmp)) fs::remove(info.existing_file_cmp);
+  if (!fs::exists(info.existing_file)) {
+    std::string cmd = "{ tr -dc '[:alnum:]' < /dev/urandom | head -c " +
+                      std::to_string(args.request_size * info.num_iterations) +
+                      "; } > " + info.existing_file + " 2> /dev/null";
+    int status = system(cmd.c_str());
+    REQUIRE(status != -1);
+    REQUIRE(fs::file_size(info.existing_file) ==
+            args.request_size * info.num_iterations);
+    info.total_size = fs::file_size(info.existing_file);
+  }
+  if (!fs::exists(info.existing_file_cmp)) {
+    std::string cmd = "cp " + info.existing_file + " " + info.existing_file_cmp;
+    int status = system(cmd.c_str());
+    REQUIRE(status != -1);
+    REQUIRE(fs::file_size(info.existing_file_cmp) ==
+            args.request_size * info.num_iterations);
+  }
+  REQUIRE(info.total_size > 0);
+#if HERMES_INTERCEPT == 1
+  INTERCEPTOR_LIST->hermes_flush_exclusion.insert(info.existing_file_cmp);
+  INTERCEPTOR_LIST->hermes_flush_exclusion.insert(info.new_file_cmp);
+#endif
+  return 0;
+}
+
+int posttest(bool compare_data = true) {
+#if HERMES_INTERCEPT == 1
+  INTERCEPTOR_LIST->hermes_flush_exclusion.insert(info.existing_file);
+  INTERCEPTOR_LIST->hermes_flush_exclusion.insert(info.new_file);
+#endif
+  if (compare_data && fs::exists(info.new_file) &&
+      fs::exists(info.new_file_cmp)) {
+    size_t size = fs::file_size(info.new_file);
+    REQUIRE(size == fs::file_size(info.new_file_cmp));
+    if (size > 0) {
+      std::vector<unsigned char> d1(size, '0');
+      std::vector<unsigned char> d2(size, '1');
+
+      FILE* fh1 = fopen(info.new_file.c_str(), "r");
+      REQUIRE(fh1 != nullptr);
+      size_t read_d1 = fread(d1.data(), size, sizeof(unsigned char), fh1);
+      REQUIRE(read_d1 == sizeof(unsigned char));
+      int status = fclose(fh1);
+      REQUIRE(status == 0);
+
+      FILE* fh2 = fopen(info.new_file_cmp.c_str(), "r");
+      REQUIRE(fh2 != nullptr);
+      size_t read_d2 = fread(d2.data(), size, sizeof(unsigned char), fh2);
+      REQUIRE(read_d2 == sizeof(unsigned char));
+      status = fclose(fh2);
+      REQUIRE(status == 0);
+
+      size_t char_mismatch = 0;
+      for (size_t pos = 0; pos < size; ++pos) {
+        if (d1[pos] != d2[pos]) char_mismatch++;
+      }
+      REQUIRE(char_mismatch == 0);
```

Comment:
Wouldn't `REQUIRE(d1 == d2)` work here?

Reply:
It would but this is better for debugging as it will say how many chars are not matching. Also, the REQUIRE will print d1 and d2 strings which are then printed in the output. So I prefer doing the comparator outside.